app/Kconfig

```diff
@@ -118,9 +94,17 @@ if ZMK_SPLIT
 
 config ZMK_SPLIT_BLE
 	bool "Split keyboard support via BLE transport"
-	depends on ZMK_BLE
```

Comment:
I don't really agree with this tweak. Can you explain why this was necessary?

Reply:
I think this came from wanting a split keyboard that only interfaced with a host via USB. Each half would need to enable BLE for connecting to the other half, but not for connecting to a host. I may have misinterpreted what I needed to do to set that up though.

Reply:
We can explore that, but I'd rather not make that also part of the refactor here. My general philosophy is:

* Merge a pure refactor, with no additions/features.
* Build your work on top of the refactor as a follow up PR.

It allows keeping focus, and also helping identify the true source of regressions. It also helps reviewers understand scope of changes much more clearly.